java/clients/src/main/java/sleeper/clients/cli/SleeperClient.java

```diff
@@ -194,6 +204,61 @@ public QueryExecutor getQueryExecutor(String tableName) {
         return executor;
     }
 
+    /**
+     * Ingests the data in the given files to the Sleeper table with name table_name. This is done by posting a
+     * message containing the list of files to the ingest queue. These files must be in S3. They can be either files or
+     * directories. If they are directories then all Parquet files under the directory will be ingested.
+     * Files should be specified in the format 'bucket/file'.
+     *
+     * @param  tableName table name to write to
+     * @param  jobId     id of the ingest job, randomly generated if not set
+     * @param  files     list of files containing records to ingest
+     * @return           id for the job create for the ingest
+     */
+    public String ingestParquetFilesFromS3(String tableName, String jobId, List<String> files) {
+        if (jobId == null) {
+            jobId = UUID.randomUUID().toString();
+        }
```

Comment:
It might be worth separating out the case where we do or do not set the job ID as separate method signatures.